diff --git a/src/core/others/ojph_arch.cpp b/src/core/others/ojph_arch.cpp
--- a/src/core/others/ojph_arch.cpp
+++ b/src/core/others/ojph_arch.cpp
@@ -193,8 +193,10 @@
              if (hwcaps & HWCAP_ASIMD) {
                level = ARM_CPU_EXT_LEVEL_ASIMD;
                if (hwcaps & HWCAP_SVE) {
                  level = ARM_CPU_EXT_LEVEL_SVE;
+#ifdef HWCAP2_SVE2
                  if (hwcaps2 & HWCAP2_SVE2)
                    level = ARM_CPU_EXT_LEVEL_SVE2;
+#endif
                }
              }
