diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -214,9 +214,7 @@
     elseif(ZLIB_FOUND AND NOT MZ_FORCE_FETCH_LIBS)
         message(STATUS "Using ZLIB ${ZLIB_VERSION}")
 
-        list(APPEND MINIZIP_INC ${ZLIB_INCLUDE_DIRS})
-        list(APPEND MINIZIP_LIB ${ZLIB_LIBRARIES})
-        list(APPEND MINIZIP_LBD ${ZLIB_LIBRARY_DIRS})
+        list(APPEND MINIZIP_LIB ZLIB::ZLIB)
 
         set(PC_PRIVATE_DEPS "zlib")
         set(ZLIB_COMPAT ON)
@@ -333,9 +331,7 @@
     if(LIBLZMA_FOUND AND NOT MZ_FORCE_FETCH_LIBS)
         message(STATUS "Using LZMA ${LIBLZMA_VERSION}")
 
-        list(APPEND MINIZIP_INC ${LIBLZMA_INCLUDE_DIRS})
-        list(APPEND MINIZIP_LIB ${LIBLZMA_LIBRARIES})
-        list(APPEND MINIZIP_LBD ${LIBLZMA_LIBRARY_DIRS})
+        list(APPEND MINIZIP_LIB LibLZMA::LibLZMA)
 
         set(PC_PRIVATE_LIBS "${PC_PRIVATE_LIBS} -llzma")
     elseif(MZ_FETCH_LIBS)
